runtime/services/catalog/artifacts/sql/sql.go

```diff
@@ -42,3 +48,42 @@ func (r *artifact) Serialise(ctx context.Context, catalogObject *drivers.Catalog
 	}
 	return catalogObject.GetModel().Sql, nil
 }
+
+var (
+	QueryCommentRegex     = regexp.MustCompile(`(?m)--.*$`)
+	MultipleSpacesRegex   = regexp.MustCompile(`\s\s+`)
+	SpacesAfterCommaRegex = regexp.MustCompile(`,\s+`)
+	MaterializedRegex     = regexp.MustCompile(`--\s*@materialize[ |\t]?:[ |\t]*([a-zA-Z]*)\s+`)
```

Comment:
1. yes it will be
2. yes
3. If we add `^` then we will to check each line as you said otherwise it will not be able to parse if the tag is not on first line. Currently we are getting input as a single string therefore I removed the `^`. I can make the change if you think so.
4.  `-- @materialize: false stuff` I expected it to be a valid comment.

Reply:
3. Maybe set the `m` flag, then it will match line separately instead. From https://pkg.go.dev/regexp/syntax: `multi-line mode: ^ and $ match begin/end line in addition to begin/end text (default false)`
4. It's fine for now, let's not worry about it